src/Generation/ResourcesGenerator.php

```diff
@@ -524,45 +524,33 @@ private static function getQueryParams(MethodDetails $method): Vector
      *
      * @return array<string, string> Fields to autopopulate in GAX if unset by user
      */
-    private static function getUuid4FieldsToAutoPopulate(
+    private static function getAutoPopulatedUuid4Fields(
         MethodDetails $method,
-        ServiceDetails $serviceDetails
+        ServiceDetails $service
     ): array {
-        $autoPopulatedFields = Vector::new([]);
-
-        $methodSettings = $serviceDetails->serviceYamlConfig->methodSettings;
-        foreach ($methodSettings as $methodSetting) {
-            $autoPopulatedFieldNames = iterator_to_array($methodSetting->getAutoPopulatedFields());
-            if (!empty($autoPopulatedFieldNames)) {
-
-                // Check if the $methodSetting is for this particular $method
-                $splitMethodName = explode('.', $methodSetting->getSelector());
-                $methodName = $splitMethodName[array_key_last($splitMethodName)];
-                if ($methodName !== $method->name) {
-                    continue;
-                }
+        $result = Map::new();
 
-                // Process only if the method is unary
-                if (!$method->isStreaming()) {
-                    $autoPopulatedFields = $method->optionalFields
-                        ->filter(
-                            // Filter out field names to autopopulate from all fields of the method
-                            fn ($f) => in_array($f->name, $autoPopulatedFieldNames)
-                        )
-                        ->filter(
-                            // Filter for optional field and uuid4 format
-                            fn ($f) => !$f->isRequired && $f->isUuid4
-                        );
-                }
-            }
+        if ($method->isStreaming()) {
+            return [];
+        }
+
+        $methodSetting = $service->serviceYamlConfig->methodSettings
+            ->filter(function ($x) use ($method) {
+                $splitName = explode('.', $x->getSelector());
+                return $method->name == $splitName[array_key_last($splitName)];
+            })->firstOrNull();
+
+        if (!is_null($methodSetting) && count($methodSetting->getAutoPopulatedFields())) {
+            $fieldNames = iterator_to_array($methodSetting->getAutoPopulatedFields());
+            $result = $method->optionalFields
+                ->filter(fn ($x) => in_array($x->name, $fieldNames))
+                ->filter(fn ($x) => $x->isUuid4)
+                ->toMap(fn ($x) => $x->camelName, fn ($x) => 'UUID4');
```

Comment:
I think you mentioned this in an earlier thread, but should we use the enum constant directly here instead of a string? We use the constant for call type here: https://github.com/googleapis/gapic-generator-php/blob/7cd894f4812f1ffdeeb4b5f35390ca9b71c395c2/src/Generation/ResourcesGenerator.php#L57

Reply:
Oh, thanks for the info.
Done